File: Swift/ExampleTests/common/ItemPriceInputString.swift
public struct ItemPriceInputString: CustomStringConvertible, Equatable {
    public let totalItems: String

    public init(totalItems: String) {
        self.totalItems = totalItems
    }

    public init(fromArray v: [String]) {
        self.totalItems = v.count > 0 ? v[0] : ""
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    public static func fromText(_ text: String) -> ItemPriceInputString {
        let parts = Tokens.require(text, 1, "ItemPriceInput")
        return ItemPriceInputString(totalItems: parts[0])
    }

    public var description: String {
        return Tokens.token(totalItems)
    }

    public static let dncString = "?DNC?"

    public static func == (a: ItemPriceInputString, b: ItemPriceInputString) -> Bool {
        return (a.totalItems == dncString || b.totalItems == dncString || a.totalItems == b.totalItems)
    }
}

File: Python/ExampleTests/common/item_price_input_string.py
from . import tokens
DNC_STRING = '?DNC?'


class ItemPriceInputString:
    def __init__(self, total_items: str = ''):
        self.total_items = total_items

    @classmethod
    def from_list(cls, values):
        v = list(values)
        return cls(
            v[0] if len(v) > 0 else ''
        )

    @classmethod
    def from_text(cls, text):
        """Builds from the text form, e.g. Money as '25 USD'."""
        parts = tokens.require(text, 1, 'ItemPriceInput')
        return cls(
            parts[0]
        )

    def __str__(self):
        return (tokens.token(self.total_items))

    def _key(self):
        return (self.total_items,)

    def __eq__(self, other):
        if not isinstance(other, ItemPriceInputString):
            return NotImplemented
        return all(a == b or a == DNC_STRING or b == DNC_STRING
                   for a, b in zip(self._key(), other._key()))

    def __hash__(self):
        return hash(self._key())

    def __repr__(self):
        return f'ItemPriceInputString({self})'
